Tests/BotTests/Interceptor/Coordinator.swift

```diff
@@ -0,0 +1,99 @@
+import Foundation
+
+private final class RecordingContext {
+    var destination: URL
+    var recordedStubs: [Interceptor.Stub] = []
+    var session: URLSession
+
+    init(destination: URL) {
+        self.destination = destination
+        self.session = URLSession(configuration: {
+            let config = URLSessionConfiguration.default
+            config.requestCachePolicy = .reloadIgnoringLocalCacheData
+            config.urlCache = nil
+            return config
+        }())
+    }
+}
+
+final class Coordinator {
+    private var stubs: [Interceptor.Stub] = []
+    private var recordingContext: RecordingContext? = nil
+
+    var isRecordingAvailable: Bool = false
```

Comment:
So can you explain how that works, we're supposed to change that to `true` when we want to switch to record mode, and then back to `false` before pushing to Git (Similar to `self.recordMode = true` on our snapshot tests)?

But that still present the risk of accidentally committing that flag with value `true` if we forget to change it back after recording, right?

Maybe have `Interceptor.stopRecording()` return a boolean, returning `true` if it was actually recording and `false` if it wasn't recording for example, then we could call `XCTestFail("Recoding mode on")` in [here](https://github.com/Babylonpartners/Wall-E/pull/14/files#diff-7aab95dc1a106f9b6906895d60608e41R283) when we call `stopRecording` from in `perform(...)` to fail any XCTest when recording is on and ensure the CI fails even if we forgot to turn `isRecordingAvailable` back to `false` after recording.

Reply:
@AliSoftware good point but what do you think about adding a `fatalError` at the end of `stopRecording` method to abort execution when we are recording instead? Returning a boolean works too but doesn't ensure that the caller will check it/won't ignore it.

```swift
fatalError("Record mode is on. Turn record mode off and re-run your test against the newly-recorded stub.")
```

Reply:
I'm not sure but I think `fatalError` might alt all the tests? Feels quite abrupt and would also prevent any cleanup potentially done after the call of `stopRecording` to be executed, be it in the end of the caller's `tearDown` implementation if that's where they call `stopRecording`, or even worse in the middle of the test if one wanted to write a test in such a manner that `stopRecording` would be called in the middle of a test case and not at the very very end...

Wondering though if we could instead do something like:
```swift
#if canImport(XCTest)
XCTestFail(...)
#endif
```
directly inside `stopRecording`'s impl?

Reply:
Addressed in https://github.com/Babylonpartners/Wall-E/pull/14/commits/d51e27ad563f84410cdbaa597e872b653086c6b1.

Reply:
I trust you tested it in practice and it indeed does the test fail when in record mode?

Reply:
I did and works as expected, the only problem is that you don't get an indication where it failed because you have a couple of layers between the assertion and the failure but fails and ensures this won't touch the API in CI as we wanted.